packages/wrangler/src/dev/miniflare.ts

```diff
@@ -1423,3 +1523,26 @@ function getImageNameFromDOClassName(
 		),
 	};
 }
+
+/**
+ * hasUnsafeBindings is a typeguard that checks whether the user has specified unsafe
+ * bindings in their Worker options
+ */
+export function hasUnsafeBindings(
+	bindings: CfWorkerInit["bindings"]
+): bindings is CfWorkerInit["bindings"] & {
+	unsafe: { bindings: CfUnsafeBinding[] };
+} {
+	const { unsafe } = bindings;
+	return !!unsafe && !!unsafe.bindings && unsafe.bindings.length > 0;
```

Comment:
NIT: I think this can just be:

```suggestion
	return unsafe?.bindings?.length > 0;
```

Reply:
Hmm, ts is unhappy with that: 
<img width="414" height="150" alt="image" src="https://github.com/user-attachments/assets/7c8102c8-d609-4723-bc08-8626e23e4acc" />